sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala

```diff
@@ -107,6 +107,11 @@ case class UsingJoin(tpe: JoinType, usingColumns: Seq[String]) extends JoinType
   override def sql: String = "USING " + tpe.sql
 }
 
+case class LateralJoin(tpe: JoinType) extends JoinType {
+  require(Seq(Inner, LeftOuter, Cross).contains(tpe), "Unsupported lateral join type " + tpe)
```

Comment:
> Currently INNER, CROSS and LEFT OUTER join types are supported with lateral join

If we can support the other join types in future, how about checking this requirement in the `AstBuilder` side?

Reply:
@maropu Postgres supports INNER, CROSS, and LEFT lateral join, and it doesn't make sense to support RIGHT OUTER and FULL OUTER lateral join. How about let's add the other two types of supported left joins: left semi and left anti here. Then lateral join types shouldn't be changed in the future.